### PR TITLE
pnnx for torch exported program

### PR DESCRIPTION
**Pull request by AtomAlpaca**
## Summary

Add pnnx support for PyTorch `torch.export.ExportedProgram` models saved with `torch.export.save()`.

```python
ep = torch.export.export(model, (x,))
torch.export.save(ep, "model.pt2")
```

Convert the exported model with:

```shell
pnnx model.pt2
```

## Changes

- Detect legacy ExportedProgram ZIP and PT2 Archive model formats.
- Parse PT2 archive metadata and JSON without additional third-party dependencies.
- Read tensor payloads with a restricted pickle implementation.
- Reconstruct parameters, buffers, constants and tensor views.
- Load PT2 graphs into the existing `pnnx::Graph`.
- Use libtorch Dispatcher schemas to order and validate operator arguments.
- Lower static and symbolic `SymInt`, `SymFloat` and `SymBool` values.
- Evaluate exported shape guards and preserve unresolved assertions as `pnnx.Assert`.
- Convert bf16 weights correctly when writing ncnn models.
- Normalize GRU and LSTM operators for the existing pnnx RNN passes.
- Reject unsupported formats, schemas, opsets, operators, devices, layouts and symbolic expressions explicitly.
- Add PT2 conversion tests for existing operators and models.
- Add archive, JSON, schema, weight, malformed-input, dynamic-shape and generated-model tests.
- Add PT2 CI coverage on Ubuntu, macOS and Windows.
- Add a PyTorch 2.6–2.12.1 producer-version test matrix.
- Document PT2 usage and supported format versions.

## Supported formats

- PyTorch 2.6–2.7 legacy ExportedProgram ZIP
- PyTorch 2.8–2.12.1 PT2 Archive v0
- Program schema major 8, minors 2, 7, 8, 14, 15, 17 and 20
- ATen opset 10

## Tests

The complete PT2 suite contains 392 tests with TorchVision enabled:

```shell
ctest --output-on-failure -L pt2
```

The producer-version regression subset contains 383 tests:

```shell
ctest --output-on-failure -L pt2-regression
```

### CODE REVIEW

File: tools/pnnx/src/pass_level2/F_conv1d.cpp [hunk 1]
```diff
@@ -64,6 +64,33 @@ pnnx.Output             output      1 0 out
...
+        return R"PNNXIR(7767517
+9 8
+pnnx.Input              input_0     0 1 input
+pnnx.Input              input_1     0 1 weight
+pnnx.Input              input_2     0 1 bias
+pnnx.Input              input_3     0 1 stride
+pnnx.Input              input_4     0 1 padding
+pnnx.Input              input_5     0 1 dilation
+pnnx.Input              input_6     0 1 groups
+aten::conv1d            op_0        7 1 input weight bias stride padding dilation groups out
```
[the remark this answers is not in the document]
Comment (AtomAlpaca) [reply tools/pnnx/src/pass_level2/F_conv1d.cpp]: Fixed in b1237ed4. The direct conv1d/2d/3d and conv_transpose1d/2d/3d level-2 patterns now capture their convolution options as constants, so the rewritten F.conv* operators receive parameters even at optlevel=0/1. Added an optlevel=0 conv1d/conv_transpose1d PNNX+ncnn numerical regression.

File: tools/pnnx/src/pt2_graph_lowering.cpp [hunk 2]
```diff
@@ -0,0 +1,1247 @@
...
+            if (!integer_can_be_parameter(arg.ai[i]))
+            {
+                error = "integer list argument is outside the pnnx parameter range";
+                return -1;
+            }
+        }
+        value = Parameter(arg.ai);
+    }
+    else if (arg.type == Pt2Argument::Float)
+        value = std::isinf(arg.f) ? Parameter(arg.f < 0 ? "-inf" : "inf") : Parameter(arg.f);
```
[the remark this answers is not in the document]
Comment (AtomAlpaca) [reply tools/pnnx/src/pt2_graph_lowering.cpp]: Fixed in 869b33ae. NaN Float and concrete SymFloat arguments now use the same explicit string representation as infinities, and generated module/forward Python emits float('nan'). Added a PT2 full_like NaN conversion and generated-Python execution regression.

File: tools/pnnx/src/main.cpp [hunk 3]
```diff
@@ -538,7 +582,26 @@ int main(int argc, char** argv)
...
+        {
+            for (size_t i = 0; i < pnnx_graph.ops.size(); i++)
+            {
+                const pnnx::Operator* op = pnnx_graph.ops[i];
+                if (op->type != "pnnx.Expression")
+                    continue;
+
+                const std::string& expr = op->params.at("expr").s;
+                const std::string expr_op = expr.substr(0, expr.find_first_of("(,"));
+                if (expr_op == "sym_ite" || expr_op == "sym_not" || expr_op == "eq" || expr_op == "ne" || expr_op == "lt" || expr_op == "le" || expr_op == "gt" || expr_op == "ge")
```
[the remark this answers is not in the document]
Comment (AtomAlpaca) [reply tools/pnnx/src/main.cpp]: Fixed in 869b33ae. PT2 expressions are now checked before pass_ncnn, and every occurrence is matched at expression-token boundaries, so nested symbolic conditionals, comparisons, and sym_not are rejected before the ncnn expander can process them. The CLI regression now uses a supported outer add around sym_ite.

File: tools/pnnx/src/main.cpp [hunk 4]
```diff
@@ -532,13 +576,41 @@ int main(int argc, char** argv)
     fprintf(stderr, "pnnx build without onnx-zero support, skip saving onnx\n");
 #endif
 
+    if (pt2)
+    {
+        const char* unsupported_expr[] = {"sym_ite", "sym_not", "eq", "ne", "lt", "le", "gt", "ge"};
```
[the remark this answers is not in the document]
Comment (AtomAlpaca) [reply tools/pnnx/src/main.cpp]: Fixed in db8277dd. PT2 now rejects and/or/xor/lshift/rshift expression tokens before pass_ncnn. Added a dynamic two-profile symbolic AND regression that verifies the explicit unsupported-ncnn-expression failure.

File: tools/pnnx/src/pt2_graph_lowering.cpp [hunk 5]
```diff
@@ -0,0 +1,1247 @@
...
+        {
+            expr = "pow";
+            aten = "aten::pow";
+        }
+        else if (node.target == "_operator.and_")
+        {
+            expr = "and";
+            aten = "aten::__and__";
+        }
+        else if (node.target == "_operator.or_")
```
[the remark this answers is not in the document]
Comment (AtomAlpaca) [reply tools/pnnx/src/pt2_graph_lowering.cpp:584]: Fixed in db8277dd. Added _operator.xor -> xor/aten::__xor__ lowering with the same SymBool/SymInt type rules as AND and OR. Torch 2.12.1's export verifier does not currently allow builtin XOR, so the lowering is covered in the existing C++ graph pilot rather than by a generated PT2 fixture.
